packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -891,35 +1224,115 @@ export class ProxyView extends AbstractUIExtension {
         const props = { ...clone.properties };
         clone.properties = props;
         const id = props["klighd.lsp.rendering.id"];
-        // OLD: changing the rendering doesn't work when renderingrefs are used
-        // props["klighd.lsp.rendering.id"] = this.getProxyId(props["klighd.lsp.rendering.id"]);
+        const proxyId = this.getProxyId(id);
+        if (ctx.decorationMap) {
+            if (props["klighd.lsp.calculated.decoration"]) {
+                props["klighd.lsp.rendering.id"] = proxyId;
+                ctx.decorationMap[proxyId] = props["klighd.lsp.calculated.decoration"];
+            } else if (ctx.decorationMap[id]) {
+                props["klighd.lsp.rendering.id"] = proxyId;
+                ctx.decorationMap[proxyId] = ctx.decorationMap[id];
+            }
+        }
 
         if (clone.type === K_POLYGON) {
             // Arrow head
-            const angle = angleOfPoint(Point.subtract(target, source));
+            const angle = angleOfPoint(Point.subtract(target, prev));
             if (props["klighd.lsp.calculated.decoration"]) {
                 // Move arrow head if actually defined
                 props["klighd.lsp.calculated.decoration"] = { ...props["klighd.lsp.calculated.decoration"], origin: target, rotation: angle };
-            } else if (ctx.decorationMap[id]) {
+            } else if (ctx.decorationMap && ctx.decorationMap[id]) {
                 // Arrow head was in rendering refs
                 props["klighd.lsp.calculated.decoration"] = { ...ctx.decorationMap[id], origin: target, rotation: angle };
             } else {
                 // Better not to show arrow head as it would be floating around somewhere
                 return [];
             }
+        } else if (clone.type === K_SPLINE) {
+            // TODO: spline type, for now just change type
+            K_POLYLINE
```

Comment:
so now you're not changing it though, this line does nothing or am I missing something?

Reply:
I left it to not have an unused import. Can be removed since I won't change it